src/stripe.common.ts

```diff
@@ -0,0 +1,489 @@
+    enumerable: false,
+    configurable: false,
+    writable: false,
+  });
+
+  this.VERSION = Stripe.PACKAGE_VERSION;
+
+  this.on = this._emitter.on.bind(this._emitter);
+  this.once = this._emitter.once.bind(this._emitter);
+  this.off = this._emitter.removeListener.bind(this._emitter);
+
+  if (
+    props.protocol &&
+    props.protocol !== 'https' &&
+    (!props.host || /\.stripe\.com$/.test(props.host))
+  ) {
+    throw new Error(
+      'The `https` protocol must be used when sending requests to `*.stripe.com`'
+    );
+  }
+
+  const agent = props.httpAgent || null;
+
+  this._api = {
+    auth: null,
+    host: props.host || DEFAULT_HOST,
+    port: props.port || DEFAULT_PORT,
+    protocol: props.protocol || 'https',
+    basePath: DEFAULT_BASE_PATH,
+    version: props.apiVersion || DEFAULT_API_VERSION,
+    timeout: Stripe._utils.validateInteger(
+      'timeout',
+      props.timeout,
+      DEFAULT_TIMEOUT
+    ),
+    maxNetworkRetries: Stripe._utils.validateInteger(
+      'maxNetworkRetries',
+      props.maxNetworkRetries,
+      0
+    ),
+    agent: agent,
+    httpClient: props.httpClient || Stripe.createNodeHttpClient(agent),
+    dev: false,
+    stripeAccount: props.stripeAccount || null,
+  };
+
+  const typescript = props.typescript || false;
+  if (typescript !== Stripe.USER_AGENT.typescript) {
+    // The mutation here is uncomfortable, but likely fastest;
+    // serializing the user agent involves shelling out to the system,
+    // and given some users may instantiate the library many times without switching between TS and non-TS,
+    // we only want to incur the performance hit when that actually happens.
+    Stripe.USER_AGENT.typescript = typescript;
+  }
+
+  if (props.appInfo) {
+    this._setAppInfo(props.appInfo);
+  }
+
+  this._prepResources();
+  this._setApiKey(key);
+
+  this.errors = _Error;
+  this.webhooks = require('./Webhooks');
+
+  this._prevRequestMetrics = [];
+  this._enableTelemetry = props.telemetry !== false;
+
+  this._requestSender = new RequestSender(
+    this,
+    Stripe.StripeResource.MAX_BUFFERED_REQUEST_METRICS
+  );
+  // Expose StripeResource on the instance too
+  // @ts-ignore
+  this.StripeResource = Stripe.StripeResource;
+}
+
+Stripe.errors = _Error;
+Stripe.webhooks = require('./Webhooks');
+
+// @ts-ignore
+Stripe.createNodeHttpClient = null;
+
+/**
+ * Creates an HTTP client for issuing Stripe API requests which uses the Web
+ * Fetch API.
+ *
+ * A fetch function can optionally be passed in as a parameter. If none is
+ * passed, will default to the default `fetch` function in the global scope.
+ */
+Stripe.createFetchHttpClient = (fetchFn: typeof fetch): typeof HttpClient => {
+  const {FetchHttpClient} = require('./net/FetchHttpClient');
+  return new FetchHttpClient(fetchFn);
+};
+
+/**
+ * Create a CryptoProvider which uses the built-in Node crypto libraries for
+ * its crypto operations.
+ */
+// @ts-ignore
+Stripe.createNodeCryptoProvider = null;
```

Comment:
Setting `createNodeCryptoProvider ` to null, overriding in `stripe.default.ts`

Reply:
what do you think about throwing a descriptive error message instead? If we mess up and customers that were using `createNodeCryptoProvider` successfully before get the wrong version of the library at least we'll give them some inclination of what happened and where to complain.

Reply:
Same for httpClient above.